### PR TITLE
Stat fix

### PR DESCRIPTION
**Pull request by rennergade**
Pairs with NaCl PR, fixes struct ordering.

### CODE REVIEW

File: src/safeposix/syscalls/fs_calls.rs [hunk 1]
```diff
@@ -422,12 +420,11 @@ impl Cage {
 
     fn _istat_helper_chr_file(inodeobj: &DeviceInode, statbuf: &mut StatData) {
         statbuf.st_dev = 5;
-        statbuf.st_nlink = inodeobj.linkcount as u64;
         statbuf.st_mode = inodeobj.mode;
+        statbuf.st_nlink = inodeobj.linkcount;
         statbuf.st_uid = inodeobj.uid;
         statbuf.st_gid = inodeobj.gid;
         //compose device number into u64
-        statbuf.__pad0 = 0;
```
Comment (JustinCappos) [inline src/safeposix/syscalls/fs_calls.rs:430]: How does this change relate?

Comment (rennergade) [reply src/safeposix/syscalls/fs_calls.rs:430]: The `lind_stat` struct that NaCl's trusted side was using to store the stat buffer mistakenly listed `nlink` as 64 bits, even though its only 32 bits. This pad seems to have been added to fix alignment when nlink is 64 bits, but is unnecessary when fixed to 32 bits.

It's being removed here in RustPOSIX to correspond to that change in NaCl.

**Review by jesings — COMMENTED** [review]
Huh... so we don't have nlink at all?

**Comment by rennergade** [comment]
@jesings I'm not sure where you're seeing that. It's moved below `mode` where it is supposed to be.

**Comment by jesings** [comment]
Oops, I'm just blind. Looks good!

**Review by youngfalcon — APPROVED** [review]

**Pull request merged**